setup.py

```diff
@@ -2,7 +2,7 @@
 
 setup(name='pyHS100',
       version='0.3.0',
-      description='Interface for TPLink HS100 Smart Plugs.',
+      description='Interface for TPLink\'s smart devices (HSxxx plugs & LBxxx bulbs)',
```

Comment:
@rytilahti Can you fix the line too long issue and merge?

Reply:
I think the hound configuration (for this project) should be fixed for the line lengths (I think something like 100 or similar what some projects use is still fine), here's what pep8 says:
```
Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the nominal line length from 80 to 100 characters (effectively increasing the maximum length to 99 characters), provided that comments and docstrings are still wrapped at 72 characters.
```
so I think we have to decide if it's worth to force <79 limitation or not, and adjust hound accordingly. My opinion lies that it's better to make code more readable and easy to work on than trying to squeeze everything to that limit.

Case in point, in this case it would be enough to strip e.g. "HSxxx" and "LBxxx", or word "smart", but I see that it would lose some information which can be useful. Opinions?

Reply:
I don't have a really strong opinion either way, but will point out that the limit is not there to encourage you to strip information, but to allow you to choose a wrapping point that looks nice when the code is seen/used/manipulated with tools that still (infortunately) use a 80 character wrapping point.  So instead of your code looking link this
```
desc="pretend the 'r' in the next word is at 79 char
acters"
```

you can make it look nice when it is wrapped

```
desc="pretend the 'r' in the next word is at 79 \
characters"
```

And whether that looks better is a matter of opinion too.

I know it's a pain. But working on other projects I have just got used to setting my IDE to 79 characters and most are smart enough wrap nicely (sorry if you don't use an IDE :-) ) and I also run flake8 and pylint locally before getting bitten by the hound.  It would be worth adding a pylint step to tox to catch this locally, I might do a PR for that later regardless.

Again, personally either way is fine for me, I don't contribute that much anyway :-)

Reply:
My personal opinion is that some of these styling recommendations are antiquated and don't necessarily produce more readable code.  I'm definitely "for" increasing the line length to something longer! I also think that if there are rules, they should be followed, hence my comment to resolve the hound's issues - to prevent someone else introducing similar errors in the future and having to explain to him/her why this particular comment can be 86 characters but theirs has to be up to 79 characters!

While the hound produces predominantly useful recommendations, I tried rubocop (I program primarily in Ruby) and the end result was a lot more unreadable code. So I'm not a huge fan of enforcing styling rules via automated means in general.

Reply:
@gollo Do you have in mind what types of tools still use that wrapping point? I mostly hear people nowadays preferring that in order to fit two files side-by-side, which however would also be done by an IDE with dynamic wrapping where needed. For documentation etc. it can also make sense to split after each sentence to keep the diffs more readable. A PR to run pylint would be appreciated though!

@kirichkov agreed and I'm definitely for (up to) 99 character limit. The idea of having automated checks (and tests for code!) is to allow also people not so familiar with the code-base to get an idea how to keep it tidy and readable. I also feel that we should enforce the chosen character limit more carefully.

When I find some more time I will try and see how easy it would be to change the limit for our tooling.

Reply:
It is nice to have for a 3 way merge, but your right most things will wrap sensibly.  As it happens I was looking at some code on my phone just after I wrote the above and it was nice to have that line length so I didn't have to zoom to really small text. That's not often and not really that compelling.  I had a little play with pylint, there's a bit of work fixing everything (or at least telling pylint to ignore stuff) so when I get a bit more time... 99 limit is fine for me, I'll probably still end up doing 79 but the occasional line will probably make use of it!

Reply:
Ok, sounds good! I will take a look and let's rise the limit to 99 as talked in pep8 when it is easily done. I will always try to target 79 too, but if it is not easily possible I will rather go over it than to do some tricks to get it under.